### PR TITLE
executor: ChecksumTableExec respects SQLKiller

### LINKED ISSUE #50651: Cancel "validating" import job is slow
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
mysql> show import jobs;
+--------+-----------------------------------------------------------------------------+--------------------+----------+------------+---------+------------------+---------------+----------------+----------------------------+----------------------------+----------+------------+
| Job_ID | Data_Source                                                                 | Target_Table       | Table_ID | Phase      | Status  | Source_File_Size | Imported_Rows | Result_Message | Create_Time                | Start_Time                 | End_Time | Created_By |
+--------+-----------------------------------------------------------------------------+--------------------+----------+------------+---------+------------------+---------------+----------------+----------------------------+----------------------------+----------+------------+
|      1 | gs://qa-workload-datasets/global-sort/data/xxx | `test`.`item_core` |      104 | validating | running | 1.041TiB         |      21826265 |                | 2024-01-23 03:17:32.473782 | 2024-01-23 03:17:34.788021 | NULL     | root@%     |
+--------+-----------------------------------------------------------------------------+--------------------+----------+------------+---------+------------------+---------------+----------------+----------------------------+----------------------------+----------+------------+
1 row in set (0.54 sec)

mysql> cancel import job 1;
(blocking for minutes)
```

And this goroutine is still running

```
1 @ 0x1c38cee 0x1c00c0d 0x1c00812 0x4b601ff 0x4341492 0x4af038d 0x4aeabc5 0x4d9fe62 0x4d9eb26 0x4d98bbe 0x2d7de9e 0x4aaf651 0x4aaf0f6 0x4aae8b1 0x4ac2896 0x3d454b7 0x4ac278b 0x4ac5de5 0x3e87cfc 0x3e87ab8 0x3e873fe 0x3e86145 0x4ac60dc 0x3e8312a 0x3e81cf3 0x3d5296d 0x1c6dc41
# labels: {"plan_digest":"", "sql_digest":"\xfe\xd6n\xd37\xd9\xc7?\f\xbf\xe8\xdc\xf8_o\xa2ߗm즚\x91N\x86d\xfbu\xef\x1a\x93\x90"}
#	0x4b601fe	github.com/pingcap/tidb/pkg/executor.(*ChecksumTableExec).Open+0x2be					/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/checksum.go:72
#	0x4341491	github.com/pingcap/tidb/pkg/executor/internal/exec.Open+0x71						/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/internal/exec/executor.go:291
#	0x4af038c	github.com/pingcap/tidb/pkg/executor.(*ExecStmt).openExecutor+0xac					/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/adapter.go:1245
#	0x4aeabc4	github.com/pingcap/tidb/pkg/executor.(*ExecStmt).Exec+0x9a4						/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/adapter.go:567
#	0x4d9fe61	github.com/pingcap/tidb/pkg/session.runStmt+0x321							/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/session/session.go:2373
#	0x4d9eb25	github.com/pingcap/tidb/pkg/session.(*session).ExecuteStmt+0x1105					/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/session/session.go:2232
#	0x4d98bbd	github.com/pingcap/tidb/pkg/session.(*session).ExecuteInternal+0x23d					/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/session/session.go:1604
#	0x2d7de9d	github.com/pingcap/tidb/pkg/util/sqlexec.ExecSQL+0xdd							/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/util/sqlexec/restricted_sql_executor.go:243
#	0x4aaf650	github.com/pingcap/tidb/pkg/executor/importer.checksumTable.func1+0x2f0					/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/importer/table_import.go:923
#	0x4aaf0f5	github.com/pingcap/tidb/pkg/executor/importer.checksumTable+0x215					/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/importer/table_import.go:950
#	0x4aae8b0	github.com/pingcap/tidb/pkg/executor/importer.VerifyChecksum+0x1d0					/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/importer/table_import.go:872
#	0x4ac2895	github.com/pingcap/tidb/pkg/disttask/importinto.postProcess.func3+0x55					/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/disttask/importinto/subtask_executor.go:119
#	0x3d454b6	github.com/pingcap/tidb/pkg/disttask/framework/storage.(*TaskManager).WithNewSession+0xd6		/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/disttask/framework/storage/task_table.go:136
#	0x4ac278a	github.com/pingcap/tidb/pkg/disttask/importinto.postProcess+0x2aa					/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/disttask/importinto/subtask_executor.go:118
#	0x4ac5de4	github.com/pingcap/tidb/pkg/disttask/importinto.(*postProcessStepExecutor).RunSubtask+0x224		/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/disttask/importinto/task_executor.go:459
#	0x3e87cfb	github.com/pingcap/tidb/pkg/disttask/framework/taskexecutor.(*BaseTaskExecutor).runSubtask.func1+0x17b	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/disttask/framework/taskexecutor/task_executor.go:338
#	0x3e87ab7	github.com/pingcap/tidb/pkg/disttask/framework/taskexecutor.(*BaseTaskExecutor).runSubtask+0x57		/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/disttask/framework/taskexecutor/task_executor.go:339
#	0x3e873fd	github.com/pingcap/tidb/pkg/disttask/framework/taskexecutor.(*BaseTaskExecutor).runStep+0xddd		/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/disttask/framework/taskexecutor/task_executor.go:319
#	0x3e86144	github.com/pingcap/tidb/pkg/disttask/framework/taskexecutor.(*BaseTaskExecutor).RunStep+0xe4		/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/disttask/framework/taskexecutor/task_executor.go:164
#	0x4ac60db	github.com/pingcap/tidb/pkg/disttask/importinto.(*importExecutor).RunStep+0x11b				/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/disttask/importinto/task_executor.go:478
#	0x3e83129	github.com/pingcap/tidb/pkg/disttask/framework/taskexecutor.(*Manager).handleExecutableTask+0x9e9	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/disttask/framework/taskexecutor/manager.go:424
#	0x3e81cf2	github.com/pingcap/tidb/pkg/disttask/framework/taskexecutor.(*Manager).handleExecutableTasks.func1+0x72	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/disttask/framework/taskexecutor/manager.go:275
#	0x3d5296c	github.com/pingcap/tidb/pkg/resourcemanager/pool/spool.(*Pool).run.func1+0x4c				/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/resourcemanager/pool/spool/spool.go:144
```

after 7 minutes, it's canceled

```
mysql> cancel import job 1;
Query OK, 0 rows affected (7 min 55.37 sec)
```

```
mysql> show import jobs\G
*************************** 1. row ***************************
          Job_ID: 1
     Data_Source: gs://qa-workload-datasets/global-sort/data/xxx
    Target_Table: `test`.`item_core`
        Table_ID: 104
           Phase: validating
          Status: cancelled
Source_File_Size: 1.041TiB
   Imported_Rows: NULL
  Result_Message: cancelled by user
     Create_Time: 2024-01-23 03:17:32.473782
      Start_Time: 2024-01-23 03:17:34.788021
        End_Time: NULL
      Created_By: root@%
1 row in set (0.20 sec)
```

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: close #50651

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by wshwsh12 — APPROVED** [review]

File: pkg/executor/checksum.go [hunk 1]
```diff
@@ -155,6 +160,9 @@ func (e *ChecksumTableExec) handleChecksumRequest(req *kv.Request) (resp *tipb.C
 			return nil, err
 		}
 		updateChecksumResponse(resp, checksum)
+		if err = e.Ctx().GetSessionVars().SQLKiller.HandleSignal(); err != nil {
```
Comment (ywqzzy) [inline pkg/executor/checksum.go:163]: I am wondering if it is needed.

Comment (ywqzzy) [reply pkg/executor/checksum.go:163]: Yes. So wired cancel mechanism

**Review by ywqzzy — APPROVED** [review]

**Pull request merged**